ledger/participant-integration-api/src/main/scala/platform/store/dao/events/PostCommitValidation.scala

```diff
@@ -105,6 +114,34 @@ private[dao] object PostCommitValidation {
           }
         )
 
+    private def validatePartiesAllocation(
+        transaction: CommittedTransaction): Option[RejectionReason] = {
+      def foldInformees[T](tx: CommittedTransaction, init: T)(
+          f: (T, String) => T
+      ): T =
+        tx.fold(init) {
+          case (accum, (_, node)) =>
+            node.informeesOfNode.foldLeft(accum)(f)
+        }
+
+      val informees = foldInformees(transaction, Seq.empty[Party]) { (informeesSoFar, partyId) =>
+        Party.assertFromString(partyId) +: informeesSoFar
+      }
+      val allocatedInformees = getPartiesSync(informees).map(_.party)
+      if (allocatedInformees.toSet == informees.toSet)
+        None
+      else
+        Some(RejectionReason.PartyNotKnownOnLedger("Some parties are unallocated"))
+    }
+
+    private def getPartiesSync(parties: Seq[Party]): List[PartyDetails] =
+      Await.result(getParties.getOrElse(identityGetParties _)(parties), AwaitDuration)
```

Comment:
I don't like this `Await` here. Can we propagate the future outwards?

Reply:
We probably can but I think it might not be easy and probably it's not worth it:

1. this change is on the previous DAML on SQL that is now only being maintained so I think it should quick, small and have the smallest impact possible
2. `PostCommitValidation.validate` was designed as sync but we can only reuse a query behind an async interface (and it doesn't make a lot of sense to create a new one)
3. `Await.result` will block the calling thread, in line with the sync/blocking nature of `PostCommitValidation.validate` so I think it's not so bad after all

Reply:
The caller is itself inside a `Future`, so it's blocking the execution context. This can be quite dangerous.

`PostCommitValidation#validate` is only called in one place (except for tests), and could easily be transformed to return a `Future[Option[RejectionReason]]`.

Reply:
This is pre-existing though and has not caused problems AFAIK but I'll refactor it, only I'd do that in a separate PR (but right after merging this, so we don't forget). Would that be OK?

Reply:
Sold!

Reply:
BTW, consider that it hasn't caused problems because up until now, there have been no `Await` calls here. In other words, it has been very fast.

Reply:
It seems to have been compute-bound so far, yes. OK, that being the case then I actually think it's better to tackle this as part of this PR.

Reply:
Nah, this too has far-reaching effects, up to the Runner. It'd better be done here but that would inflate this PR too much.

Reply:
Created #7972.